nntrainer/graph/graph_core.cpp

```diff
@@ -182,8 +182,16 @@ void GraphCore::ensureName(GraphNode &node, const std::string &prefix,
 
 void GraphCore::removeLastNode() {
   auto last_node = Sorted.back();
+
   Sorted.pop_back();
-  adj.erase(adj.begin() + last_node->getIndex());
+
+  /// fix index after last node has been deleted.
+  auto after = adj.erase(adj.begin() + last_node->getIndex());
+  auto after_index = last_node->getIndex();
+
+  for (; after != adj.end(); ++after) {
+    after->front()->setIndex(after_index++);
```

Comment:
can you please elaborate why the index of the elements after the current elements needs updating? The index of a node a unique identifier of a node in the graph. Removing a node still maintains the identifier to be unique.

Reply:
I get, this updates the index of each node to ensure the invariant that `adj[i].front().getIndex() == i`.